roles/edpm_nova/tasks/configure.yml

```diff
@@ -118,3 +118,70 @@
     mode: '0600'
     owner: nova
     group: nova
+
+- name: Check if compute_id exists in state_path
+  tags:
+    - configure
+    - nova
+    - adoption
+  become: true
+  ansible.builtin.stat:
+    path: /var/lib/nova/compute_id
+  register: old_compute_id
+
+- name: Copy compute_id from state_path to the config dir if present
+  tags:
+    - configure
+    - nova
+    - adoption
+  become: true
+  ansible.builtin.copy:
+    src: "/var/lib/nova/compute_id"
+    remote_src: true
+    dest: "{{ edpm_nova_config_dest }}/compute_id"
+    owner: nova
+    group: nova
+    mode: "0500"
+  when: old_compute_id.stat.exists
+
+- name: Delete compute_id from state_path
+  tags:
+    - configure
+    - nova
+    - adoption
+  become: true
+  ansible.builtin.file:
+    state: absent
+    path: "/var/lib/nova/compute_id"
+  when: old_compute_id.stat.exists
+
+- name: Check if compute_id exists in the config dir
+  tags:
+    - configure
+    - nova
+  become: true
+  ansible.builtin.stat:
+    path: "{{ edpm_nova_config_dest }}/compute_id"
+  register: compute_id
+
+- name: Pre-generate compute_id if not already exists
+  tags:
+    - configure
+    - nova
+  become: true
+  ansible.builtin.shell:
+    cmd: uuidgen > "{{ edpm_nova_config_dest }}/compute_id"
+  changed_when: true
+  when: not compute_id.stat.exists
+
+- name: Make compute_id ready only
+  tags:
+    - configure
+    - nova
+  become: true
+  ansible.builtin.file:
+    path: "{{ edpm_nova_config_dest }}/compute_id"
+    state: file
+    owner: nova
+    group: nova
+    mode: "0400"
```

Comment:
I would also call ``chattr +i`` in shell

Reply:
Good suggestion. Done.